x-pack/plugins/osquery/server/types.ts

```diff
@@ -26,10 +31,14 @@ export interface SetupPlugins {
   data: DataPluginSetup;
   features: PluginSetupContract;
   security: SecurityPluginStart;
+  taskManager?: TaskManagerPluginSetup;
+  telemetry?: TelemetryPluginSetup;
 }
 
 export interface StartPlugins {
   actions: ActionsPlugin['start'];
   data: DataPluginStart;
   fleet?: FleetStartContract;
+  taskManager?: TaskManagerPluginStart;
```

Comment:
I think you can drop this if you are going down the soft real-time route with the buffer/sender.

Reply:
we have discussed it and it seems an interval-based solution will better fit our needs